backend/internal/storage/postgres/schema/playlist.go

```diff
@@ -0,0 +1,59 @@
+package schema
+
+import (
+	"context"
+	"platnm/internal/models"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+type PlaylistRepository struct {
+	*pgxpool.Pool
+}
+
+func (r *PlaylistRepository) CreatePlaylist(ctx context.Context, playlist models.Playlist) error {
+
+	query := `
+	INSERT INTO playlist (title, user_id, bio, cover_photo)
+	VALUES ($1, $2, $3, $4);
+	`
+
+	_, err := r.Exec(ctx, query, playlist.Title, playlist.UserID, playlist.Bio, playlist.CoverPhoto)
+
+	if err != nil {
+		return err
+	}
+	return nil
+
+}
+
+func (r *PlaylistRepository) AddToUserOnQueue(ctx context.Context, id string, track models.Track) error {
+
+	var playlistID string
+
+	findQuery := `SELECT id FROM playlist WHERE user_id = $1 AND title = 'On Queue'`
+	err := r.QueryRow(ctx, findQuery, id).Scan(&playlistID)
+
+	if err != nil {
+		return err
+	}
+
+	// now insert the track into the user's on queue playlist
+	insertQuery := `
+	INSERT INTO playlist_track (playlist_id, track_id)
```

Comment:
what happens if the track id is not a real track we have? 👀
i fear you might have a foreign key error, you can look at the CreateComment method for inspo on handling it

Reply:
fixed this also! here is the screenshot of a postman call with a not real id!
<img width="781" alt="Screenshot 2024-10-25 at 11 59 18 PM" src="https://github.com/user-attachments/assets/05954df2-d02a-44ae-9cdb-26dbf554be3c">